backend/infrahub/permissions/report.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from infrahub.core.account import fetch_permissions
+from infrahub.core.constants import GlobalPermissions, PermissionDecision
+from infrahub.core.registry import registry
+from infrahub.permissions.local_backend import LocalPermissionBackend
+
+if TYPE_CHECKING:
+    from infrahub.core.schema import MainSchemaTypes
+    from infrahub.database import InfrahubDatabase
+    from infrahub.graphql import GraphqlContext
+    from infrahub.permissions.types import KindPermissions
+
+
+async def report_schema_permissions(
+    db: InfrahubDatabase, schemas: list[MainSchemaTypes], context: GraphqlContext
+) -> list[KindPermissions]:
```

Comment:
I think it would be better to remove the `GraphqlContext` dependency from this function b/c it is not in the `graphql` directory and could be used outside of a GraphQL context. It would be cleaner to have the function accept the account session and branch as individual parameters

Reply:
Yes good point. It was within the graphql folder to begin with but I felt like it was cluttering up things there a bit. I think we'll want to send in the service object here eventually and have access to the cache. But I've swapped out context -> branch and account_session here for now.